scripts/globals/events/egg_hunt_egg-stravaganza.lua

```diff
@@ -0,0 +1,852 @@
+-- ----------------------------------
+-- Egg Hunt Egg-Stravaganza
+-- ----------------------------------
+require("scripts/globals/settings")
+require("scripts/globals/utils")
+require("scripts/globals/npc_util")
+-- ----------------------------------
+xi = xi or {}
+xi.events = xi.events or {}
+xi.events.egg_hunt = xi.events.egg_hunt or {}
+xi.events.egg_hunt.data = xi.events.egg_hunt.data or {}
+xi.events.egg_hunt.entities = xi.events.egg_hunt.entities or {}
+
+local event = SeasonalEvent:new("egg_hunt")
+
+-- Default settings
+local settings =
+{
+    START  = { DAY   =  6, MONTH = 4 },
+    FINISH = { DAY   = 17, MONTH = 4 },
+
+    VAR =
+    {
+        DAILY_EGG     = "[EGGHUNT]DAILY_EGG",
+        DAILY_REWARD  = "[EGGHUNT]DAILY_REWARD",
+        DAILY_BONUS   = "[EGGHUNT]DAILY_BONUS",
+        DAILY_HELM    = "[EGGHUNT]DAILY_HELM",
+        INITIAL_THREE = "[EGGHUNT]INITIAL_THREE",
+    },
+
+    -- Default allows additional eras to be added each year
+    ERA_2007 = false, -- Jeweled Egg and Egg Helm
+    ERA_2008 = false, -- Tier 2 nation eggs
+    ERA_2009 = false, -- Egg Buffet set
+
+    -- Consolation prizes for repeating combinations where
+    -- the player has already received the relevant reward
+    MINOR_REWARDS = true,
+
+    -- Don't add words here
+    -- settings/main.lua will overwrite these
+    -----------------------------------------------
+    -- Set custom combinations, eg. WORD = 1234
+    -- Where WORD is an arrangement of lettered eggs
+    -- Where 1234 is the itemID for the reward
+    BONUS_WORDS = {},
+}
+
+local function loadSettings(tbl, name)
+    if not xi.settings.main[name] then
+        return
+    end
+
+    for k, v in pairs(tbl) do
+        if xi.settings.main[name][k] then
+            tbl[k] = xi.settings.main[name][k]
+        end
+    end
+end
+
+loadSettings(settings, "EGG_HUNT")
+
+xi.events.egg_hunt.enabledCheck = function()
+    local month = tonumber(os.date("%m"))
+    local day = tonumber(os.date("%d"))
+
+    if month == settings.START.MONTH then
+        if day >= settings.START.DAY then
+            return true
+        end
+
+    elseif month == settings.FINISH.MONTH then
+        if day <= settings.FINISH.DAY then
+            return true
+        end
+    end
+
+    return false
+end
+
+event:setEnableCheck(xi.events.egg_hunt.enabledCheck)
+
+-- ----------------------------------
+-- Data
+-- ----------------------------------
+
+xi.events.egg_hunt.data =
+{
+    [xi.zone.SOUTHERN_SAN_DORIA] =
+    {
+        id = 17719704,
```

Comment:
Hardcoded NPC IDs need to be using the zone ID lua instead for ID shifts

Reply:
You can look up and use offsets like this: https://github.com/LandSandBoat/server/blob/9d1098b4d9ebd45cb66b92201c37351bcc9f5507/scripts/zones/Batallia_Downs/IDs.lua#L112

Reply:
Also the message IDs should be added to each zone, so they're included in UpdateExtractor runs